src/wallet/walletdb.cpp

```diff
@@ -110,6 +110,17 @@ bool CWalletDB::WriteMasterKey(unsigned int nID, const CMasterKey& kMasterKey)
     return Write(std::make_pair(std::string("mkey"), nID), kMasterKey, true);
 }
 
+bool CWalletDB::WriteZKey(const libzcash::PaymentAddress& addr, const libzcash::SpendingKey& key, const CKeyMetadata &keyMeta)
+{
+    nWalletDBUpdated++;
+
+    if (!Write(std::make_pair(std::string("zkeymeta"), addr), keyMeta))
+        return false;
+
+    // pair is: tuple_key("spendingkey", paymentaddress) --> secretkey
+    return Write(std::make_pair(std::string("zkey"), addr), key, false);
```

Comment:
Is there likely to be any wallet-loading benefit to storing `Hash(pubkey, privkey)` like is done for unencrypted transparent addresses? We don't currently verify that the spending key matches the payment address; would this be cheaper than doing a direct calculation (like is the case for transparent addresses)?